charts/templates/ingress.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Release.Name }}
+  annotations:
+    {{ .Values.ingress.annotations }}
```

Comment:
I think you might need something like: 
```
{{- with .Values.ingress.annotions }}
  annotations: {{ toYaml . | nindent 4 }}
  {{- end }}

Reply:
Can you adapt the functional test as well to install an nginx ingress controller and test this endpoint instead of the current port forwarding? 

You should be able to just add a helm command that replicates the functionality of azimuth-ops:
https://github.com/stackhpc/ansible-collection-azimuth-ops/blob/500e98bc4ff8c93e8a75207f9908e483a737c866/roles/ingress_nginx/tasks/main.yml#L3-L14